core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -367,46 +368,67 @@ public boolean tableExists(SessionContext context, TableIdentifier identifier) {
   }
 
   private LoadTableResponse loadInternal(
-      SessionContext context, TableIdentifier identifier, SnapshotMode mode) {
+      SessionContext context,
+      TableIdentifier identifier,
+      SnapshotMode mode,
+      Map<String, Object> viewContext) {
     Endpoint.check(endpoints, Endpoint.V1_LOAD_TABLE);
     AuthSession contextualSession = authManager.contextualSession(context, catalogAuth);
+
+    // inject the context to query params
+    Map<String, String> queryParams = Maps.newHashMap(mode.params());
+
+    Object viewIdentifierObj = viewContext.get(ContextAwareTableCatalog.VIEW_IDENTIFIER_KEY);
+    if (viewIdentifierObj instanceof String) {
```

Comment:
If there is a view identifier provided here but it's not string, or it has less than 2 parts we silently omit the provided input. Shouldn't we raise an illegal argument exception rather to tell the clients that whatever they provided was incorrect?

Reply:
+1